inc/namespace.php

```diff
@@ -47,6 +48,11 @@ function redirect_user() {
 		return;
 	}
 
+	if ( $_SERVER['REQUEST_URI'] && in_array( $_SERVER['REQUEST_URI'], $allowed, true ) ) {
```

Comment:
@robindevitt I think this would be more readable if done like the `oauth2` override at the top, also `$allowed` is expected to have specific allowed values for `$page`, so it doesn't feel natural to treat `/robots.txt`, as this condition says: 
- for any requests where `$page` is not defined but the request URI is added to `$allowed` array, add the filter and return.

I think something along the lines of the following, added to the top, would be okay:
```php
if ( ( $_SERVER['REQUEST_URI'] ?? '' ) === '/robots.txt' ) {
	add_filter( 'robots_txt', __NAMESPACE__ . '\\create_robots_file' );
	return;
}
```